app/scripts/modules/openstack/common/cacheBackedMultiSelectField.directive.js

```diff
@@ -31,22 +31,37 @@ module.exports = angular.module('spinnaker.openstack.cacheBackedMultiSelectField
           onSelectionsChanged: function() {
             //Hack to work around bug in ui-select where selected values re-appear in the drop-down
             scope.state.selectedOptions = _.uniq(scope.state.selectedOptions);
-            scope.model = scope.state.selectedOptions;
-            if( scope.onChange ) {
-              var args = { selection: scope.model };
-              scope.onChange(args);
-            }
           }
         });
 
         function updateOptions() {
           scope.options = _.sortBy(scope.cache, 'name');
         }
 
+        //update the options whenever the associated cache gets updated
         scope.$watch('cache', updateOptions);
+
+        //The following watchers are needed to overcome the fact that ui-select requires the model to be a child
+        //property.  This code effectively removes that constraint for users of this directive.
+
+        //carry through any changes from the UI to the provided model
+        scope.$watch('state.selectedOptions', function() {
+            //make sure the update is coming from the UI and not from the parent
+            if( scope.model != scope.state.selectedOptions ) {
```

Comment:
Please reformat as `if (scope.model !== scope.state.selectedOptions) {` here and for the other if statements below.